community/io/src/main/java/org/neo4j/io/fs/FileUtils.java

```diff
@@ -340,6 +343,86 @@ public static File path( File root, String... path )
     }
 
     /**
+     * Attempts to discern if the given path is mounted on a device that can likely sustain a very high IO throughput.
+     * <p>
+     * A high IO device is expected to have negligible seek time, if any, and be able to service multiple IO requests
+     * in parallel.
+     *
+     * @param pathOnDevice Any path, hypothetical or real, that once fully resolved, would exist on a storage device
+     * that either supports high IO, or not.
+     * @param defaultHunch The default hunch for whether the device supports high IO or not. This will be returned if
+     * we otherwise have no clue about the nature of the storage device.
+     * @return Our best-effort estimate for whether or not this device supports a high IO workload.
+     */
+    public static boolean highIODevice( Path pathOnDevice, boolean defaultHunch )
```

Comment:
I didn't write automated tests for it because of how environment specific it is. We can try to stage replica environments with e.g. JimFS, though.